src/test/java/com/example/demo/controller/GreetingControllerTests.java

```diff
@@ -9,11 +10,12 @@
 
 class GreetingControllerTests {
 
+@InjectMocks
+GreetingService serviceMock;
 	@Test
 	void greetingEndpoint() {
 		String name = "Mary";
-		GreetingService ServiceMock = Mockito.mock(GreetingService.class);
-		Mockito.when(ServiceMock.printHello(name)).thenReturn("Hello "+name+"!");
-		assertEquals(new GreetingController(ServiceMock).hello(name), "Hello " + name + "!");
+		Mockito.when(serviceMock.printHello(name)).thenReturn("Hello "+name+"!");
+		assertEquals(new GreetingController(serviceMock).hello(name), "Hello " + name + "!");
```

Comment:
Mocked service is supposed to be defined as a class-member field somehow like this:

```
@Mock
private GreetingService greetingService;
```